src/qibolab/pulses.py

```diff
@@ -363,6 +363,77 @@ def __repr__(self):
         return f"{self.name}({format(self.rel_sigma, '.6f').rstrip('0').rstrip('.')})"
 
 
+class GaussianSquare(PulseShape):
+    r"""GaussianSquare pulse shape.
+
+    Args:
+        rel_sigma (float): relative sigma so that the pulse standard deviation (sigma) = duration / rel_sigma
+        width (float): Percentage of the pulse that is flat
+
+    .. math::
+
+        A\exp^{-\frac{1}{2}\frac{(t-\mu)^2}{\sigma^2}}[Rise] + Flat + A\exp^{-\frac{1}{2}\frac{(t-\mu)^2}{\sigma^2}}[Decay]
+    """
+
+    def __init__(self, rel_sigma: float, width: float):
+        self.name = "GaussianSquare"
+        self.pulse: Pulse = None
+        self.rel_sigma: float = float(rel_sigma)
+        self.width: float = float(width)
+
+    def __eq__(self, item) -> bool:
+        """Overloads == operator."""
+        if super().__eq__(item):
+            return self.rel_sigma == item.rel_sigma and self.width == item.width
+        return False
+
+    def envelope_waveform_i(self, sampling_rate=SAMPLING_RATE) -> Waveform:
+        """The envelope waveform of the i component of the pulse."""
+
+        if self.pulse:
+
+            def gaussian(t, rel_sigma, gaussian_samples):
+                mu = (2 * gaussian_samples - 1) / 2
+                sigma = (2 * gaussian_samples) / rel_sigma
+                return np.exp(-0.5 * ((t - mu) / sigma) ** 2)
```

Comment:
Maybe this could be moved outside as a module method to be reused in other shapes (for example the `Gaussian`), if applicable. This is more for @AleCandido refactoring.

Reply:
For sure, but I was also thinking to use something like:
https://scipy.github.io/devdocs/reference/generated/scipy.signal.windows.gaussian.html#scipy.signal.windows.gaussian
(though I'm not sure whether I want to add a dependency on SciPy just for this: it's only SciPy, but it's still a further dependency...)